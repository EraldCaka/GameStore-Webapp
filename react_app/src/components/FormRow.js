const FormRow = ({ type, name, value, handleChange }) => {
  const inputId = `${name}-input`;

  return (
    <div className="form-row">
      <label htmlFor={inputId} className="form-label">
        {name}
      </label>
      <input
        type={type}
        value={value}
        id={inputId}
        name={name}
        onChange={handleChange}
        className="form-input"
      />
    </div>
  );
};

export default FormRow;
