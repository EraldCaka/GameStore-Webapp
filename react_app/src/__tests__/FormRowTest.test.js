import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import FormRow from "../components/FormRow";

test("renders FormRow component", () => {
  const handleChange = jest.fn();
  const name = "name";
  const type = "text";
  const value = "john.doe";

  const { container, getByLabelText } = render(
    <FormRow
      name={name}
      type={type}
      value={value}
      handleChange={handleChange}
    />
  );

  const inputElement = getByLabelText(name);

  expect(inputElement).toBeDefined();
  expect(inputElement).toHaveAttribute("type", type);
  expect(inputElement).toHaveAttribute("name", name);

  fireEvent.change(inputElement, { target: { value: "new-value" } });

  expect(handleChange).toHaveBeenCalledTimes(1);
  expect(handleChange).toHaveBeenCalledWith(expect.any(Object));

  expect(container.firstChild).toMatchSnapshot();
});
