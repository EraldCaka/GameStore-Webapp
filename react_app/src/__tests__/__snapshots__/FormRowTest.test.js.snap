// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FormRow component 1`] = `
<div
  class="form-row"
>
  <label
    class="form-label"
    for="name-input"
  >
    name
  </label>
  <input
    class="form-input"
    id="name-input"
    name="name"
    type="text"
    value="john.doe"
  />
</div>
`;
